docker/Jenkinsfile-build-hadoop-docker

```diff
@@ -132,3 +131,63 @@ parallel(pipelineContext.getBuildConfig().getSupportedHadoopDistributions().coll
         }
     ]
 })
+
+node (pipelineContext.getBuildConfig().getDefaultNodeLabel()) {
+    final String buildStageName = "Build Edge node for 0xd cluster"
+    final String baseImageName = "${IMAGE_NAME_PREFIX}-hdp-2.2"
+    final String imageName = "${baseImageName}-0xd-edge"
+    stage(buildStageName) {
+        try {
+            pipelineContext.getBuildSummary().addStageSummary(this, buildStageName, '')
+            pipelineContext.getBuildSummary().setStageDetails(this, buildStageName, env.NODE_NAME, env.WORKSPACE)
+            cleanWs()
+
+            final boolean conflict = pipelineContext.getUtils()
+                    .dockerImageExistsInRegistry(this, params.dockerRegistry, imageName, version)
+            if (conflict && !params.force) {
+                error "Tag ${imageName}:${version} already exists in the repository"
+            }
+
+            withCredentials([file(credentialsId: 'jenkins-ldif', variable: 'JENKINS_LDIF_PATH'),
+                             usernamePassword(credentialsId: "${params.dockerRegistry}", usernameVariable: 'REGISTRY_USERNAME', passwordVariable: 'REGISTRY_PASSWORD')]) {
+                dir("${imageName}.${version}") {
+                    pipelineContext.getUtils().unstashFiles(this, DOCKER_STASH)
+                    sh """
+                                    cd docker
+
+                                    docker build \
+                                        -t ${REGISTRY_PREFIX}/${imageName}:${version} \
+                                        -f hadoop/slave/Dockerfile \
+                                        --build-arg PATH_PREFIX=hadoop/slave \
```

Comment:
I think "slave" is a misleading name, could it be edge_node instead or similar?

Reply:
was not happy about the name too, will fix